src/libs/hooks/useWindowResize.js

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState, useCallback } from 'react';
+
+export default function useWindowResize(handler) {
```

Comment:
it should be something similar to useEffect, the implementation depend of when should listen(generally when something is open). 

Possible API could be

useWindowResize(handler, isListening = true);

so it can be used to listen always or on and off

always:

useWindowResize(() => something()};

on and off

const [isOpen, setOpenState] = useState(false);
useWindowResize(() => setOpenState(false), isOpen);